api/v1alpha1/values_types.go

```diff
@@ -0,0 +1,670 @@
+	Stackdriver *TracerStackdriverConfig `json:"stackdriver,omitempty"`
+}
+
+// Configuration for the datadog tracing service.
+type TracerDatadogConfig struct {
+	// Address in host:port format for reporting trace data to the Datadog agent.
+	Address string `json:"address,omitempty"`
+}
+
+// Configuration for the lightstep tracing service.
+type TracerLightStepConfig struct {
+	// Sets the lightstep satellite pool address in host:port format for reporting trace data.
+	Address string `json:"address,omitempty"`
+	// Sets the lightstep access token.
+	AccessToken string `json:"accessToken,omitempty"`
+}
+
+// Configuration for the zipkin tracing service.
+type TracerZipkinConfig struct {
+	// Address of zipkin instance in host:port format for reporting trace data.
+	//
+	// Example: <zipkin-collector-service>.<zipkin-collector-namespace>:941
+	Address string `json:"address,omitempty"`
+}
+
+// Configuration for the stackdriver tracing service.
+type TracerStackdriverConfig struct {
+	// enables trace output to stdout.
+	Debug bool `json:"debug,omitempty"`
+	// The global default max number of attributes per span.
+	MaxNumberOfAttributes uint32 `json:"maxNumberOfAttributes,omitempty"`
+	// The global default max number of annotation events per span.
+	MaxNumberOfAnnotations uint32 `json:"maxNumberOfAnnotations,omitempty"`
+	// The global default max number of message events per span.
+	MaxNumberOfMessageEvents uint32 `json:"maxNumberOfMessageEvents,omitempty"`
+}
+
+// Configuration for base component
+type BaseConfig struct {
+	// For Helm2 use, adds the CRDs to templates.
+	EnableCRDTemplates bool `json:"enableCRDTemplates,omitempty"`
+	// URL to use for validating webhook.
+	ValidationURL string `json:"validationURL,omitempty"`
+	// For istioctl usage to disable istio config crds in base
+	EnableIstioConfigCRDs bool `json:"enableIstioConfigCRDs,omitempty"`
+	ValidateGateway       bool `json:"validateGateway,omitempty"`
+}
+
+type IstiodRemoteConfig struct {
+	// URL to use for sidecar injector webhook.
+	InjectionURL string `json:"injectionURL,omitempty"`
+	// Path to use for the sidecar injector webhook service.
+	InjectionPath string `json:"injectionPath,omitempty"`
+}
+
+type Values struct {
+	Cni    *CNIConfig    `json:"cni,omitempty"`
+	Global *GlobalConfig `json:"global,omitempty"`
+	Pilot  *PilotConfig  `json:"pilot,omitempty"`
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// +kubebuilder:validation:Schemaless
+	Ztunnel map[string]string `json:"ztunnel,omitempty"`
+	// Controls whether telemetry is exported for Pilot.
+	Telemetry              *TelemetryConfig       `json:"telemetry,omitempty"`
+	SidecarInjectorWebhook *SidecarInjectorConfig `json:"sidecarInjectorWebhook,omitempty"`
+	IstioCni               *CNIConfig             `json:"istio_cni,omitempty"`
+	Revision               string                 `json:"revision,omitempty"`
+	OwnerName              string                 `json:"ownerName,omitempty"`
```

Comment:
This is used for the `install.operator.istio.io/owning-resource` label. Should we remove this value from the CRD and set it to some fixed value when we render the charts?